tutorials/haskell/gui-application/tutorial.md

```diff
@@ -0,0 +1,825 @@
+---
+title: Creating a GUI application in Haskell
+published: 2016-06-29
+update: 2016-06-29
+ghc: 7.10.3
+lts: 5.18
+libraries: gtk3-0.14.4 glib-0.13.2.2
+language: haskell
+author: Mark
+author-name: Mark Karpov
+---
+
+This tutorial shows how to build a GUI application in Haskell using bindings
+to GTK+. While working on a calculator program we'll cover he following
+topics:
+
+* GTK+ terminology
+* Writing handlers that process user's actions (button click, etc.)
+* Introducing global state
+* Manual creation of forms and creation of forms using Glade
+
+Once you finish with the tutorial you will have solid understanding how to
+move on, read the documentation of the
+[`gtk3`](https://hackage.haskell.org/package/gtk3) package and accomplish
+your tasks.
+
+The tutorial does not assume any knowledge of Haskell except for very basic
+understanding of how to work with `IO` monad. The GTK+ binding is very
+straightforward and imperative in its nature. This may be seen as a
+downside, but I think it also may make things easier for newcomers to
+Haskell programming with imperative background.
```

Comment:
@mrkkrp 

- "except for **a** very basic understanding of how to work with **the** `IO` monad."
- "Haskell programming with **an** imperative background."